ci: update site-deploy.yml

Wxh16144 opened the pull request:
<!--
First of all, thank you for your contribution! 😄

For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.

Before submitting your pull request, please make sure the checklist below is confirmed.

Your pull requests will be merged after one of the collaborators approve.

Thank you!

-->

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Other (about what?)

### 🔗 Related issue link

close: #39303

<!--
1. Put the related issue or discussion links here.
-->

### 💡 Background and solution

- Modify trigger events
- Add Surge Deployment
- Add Deployment Success Comment function

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list the final API implementation and usage sample if that is a new feature.
-->

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

Wxh16144 committed:
ci: add doc-site workflow

Wxh16144 committed:
ci: Add Manual Trigger Action

MadCcc commented on .github/workflows/doc-site.yml:
```diff
@@ -0,0 +1,66 @@
+
+name: Doc Site
+on:
+  release:
+    types: [published]
```
监听 tag 好点？

Wxh16144 replied:
嗯嗯, 我修改一下。
不过下面的 steps 我只测试了后半部分，前半部分调试赐教一下?

afc163 commented:
加到 https://github.com/ant-design/ant-design/blob/master/.github/workflows/site-deploy.yml 吧，我在这个 PR 加了标签不会自动关闭了。

afc163 commented:
https://github.com/ant-design/ant-design/blob/master/.github/workflows/site-deploy.yml 感觉也实现的不太好，on create 然后再去判断类型，这样所有操作都会触发这个 action，不太必要。

Wxh16144 committed:
ci: update site-deploy.yml

Modify trigger events

Add Surge Deployment

Add Deployment Success Comment function

Wxh16144 committed:
Revert "ci: Add Manual Trigger Action"

This reverts commit 5bd75d1e5876f38197fe20ce28e99712b5f32c8b.

Wxh16144 committed:
Revert "ci: add doc-site workflow"

This reverts commit ec7e7e40e742d9feab4d0c03303ef1e6467ee2bc.

MadCcc commented on .github/workflows/site-deploy.yml:
```diff
@@ -2,15 +2,16 @@
 jobs:
   setup:
     runs-on: ubuntu-latest
-    if: github.event.ref_type == 'tag' && (contains(github.event.ref, '-') == false)
```
(contains(github.event.ref, '-') == false)
这个还是要的，不然测试版本也会部署站点

Wxh16144 replied:
噢对，测试版的 tag。 v5.0.0-rc、v5.0.0-beta.1 等这些

Wxh16144 committed:
ci: update ci config
